pkgs/os-specific/darwin/bartender/default.nix

```diff
@@ -1,48 +1,31 @@
 { lib
-, stdenvNoCC
 , fetchurl
-, _7zz
+, darwin
 }:
 
-stdenvNoCC.mkDerivation (finalAttrs: {
+darwin.installBinaryPackage rec {
```

Comment:
thought it was preferred to use the finalAttrs pattern over rec, why switch?

Reply:
Based on my understanding, [`mkDerivation`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix#L46) takes either an attrset or a function (the `finalAttrs` pattern) as argument.

This PR introduces `installBinaryPackage` to simplify the process of packaging macOS apps and reduces the overuse of `mkDerivation` (code duplication in general). It takes an attrset as argument, i.e. the `finalAttrs` pattern is not applicable here.

Reply:
Makes sense, thanks.

Reply:
> Based on my understanding, [`mkDerivation`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix#L46) takes either an attrset or a function (the `finalAttrs` pattern) as argument.
> 
> This PR introduces `installBinaryPackage` to simplify the process of packaging macOS apps and reduces the overuse of `mkDerivation` (code duplication in general). It takes an attrset as argument, i.e. the `finalAttrs` pattern is not applicable here.

I don't think so.

The reason why most build helpers in Nixpkgs don't support fixed-point attributes (`finalAttrs: { ... }`) yet, is because of the challenge to handle technical debts inside existing build helpers -- arguments that cannot be passed directly to `stdenv.mkDerivation`.

The goal of #234651 is to introduce a way (`lib.extendMkDerivation`) to handle such arguments and bring fixed-point arguments support to existing builders. Even then, there are still efforts to bring fixed-point arguments support to build helpers like `buildRustPackage` (#194475) and `buildGoModule` (#225051).

As a new build helper, `installBinaryPackage` could avoid such technical debt as a whole, support fixed-point attributes without the need of the WIP `lib.extendMkDerivation`. We just need to make sure that everything is passed properly into `stdenv.mkDerivation`.

Reply:
Update:

Sorry @stepbrobd for my misunderstanding your point. You're right about the current implementation of `installBinaryPackage` handling only a plain attribute set.

I just want to point out that such support is not impossible or unsuitable outside `stdenv.mkDerivation`. We only need to modify the implementation of `installBinaryPackage` a bit to support it.